packages/application-shell/src/test-utils/test-utils.js

```diff
@@ -160,29 +160,33 @@ const render = (
 const renderWithRedux = (
   ui,
   {
-    // Consumers of renderWithRedux can use
-    //   { store: createReduxStore({ requestsInFlight: null, .. }) }
-    // to pass an initial state to Redux.
-    store = createReduxStore(),
+    // Pass an initial state that will be used to rehydrate the redux store.
+    // Possible state values are:
+    // - requestsInFlight: []
+    // - notifications: []
+    preloadedState,
```

Comment:
With this change, we gain a simpler API to write a test, but we lose the ability to pass in an arbitrary store. Which means we can't replace the middleware for our tests anymore. I'm not sure whether we should already limit the API that much. Did we learn that nobody ever needs to pass in a configured store?

Reply:
I thought about it but I don't know when we would be needing to replace the `store`. But I can revert this change if you prefer so.

Reply:
I generally like to keep the API open until the use-cases are clear. Then we can restrict the API to force people in the right direction. I can't make a good call here at the moment.

Are all stores configured the same for all applications? If yes, then it makes sense to configure the store in a similar way for our tests and we might not need to enable injecting middleware. It's still an assumption though.

I'd keep it open for now until we've seen more cases using it. If they all end up being the same (using preloaded state only), we can limit the API.

Reply:
I reverted the change